plan/plans/show.go

```diff
@@ -356,8 +356,16 @@ func (s *ShowPlan) fetchShowTables(ctx context.Context) error {
 			s.Pattern.Expr = expression.Value{Val: data[0]}
 		} else if s.Where != nil {
 			m[expression.ExprEvalIdentFunc] = func(name string) (interface{}, error) {
-				if s.Full && strings.EqualFold(name, "Table_type") {
-					return data[1], nil
+				// The first column is Tables_in_{database}.
+				// If s.Full is true, there is a column named Table_type at the second place.
+				if strings.EqualFold(name, "Table_type") {
```

Comment:
the if check seems the same as `if s.Full && strings.EqualFold(name, "Table_type")`, why use two if check?